[SR-7904] Add source filenames in the output of `-dump-ast`

fedetrim opened the pull request:
<!-- What's in this pull request? -->
Text by Vinicius Vendramini in [SR-7904](https://bugs.swift.org/browse/SR-7904):

The Swift compiler's `-dump-ast` functionality prints an output that doesn't include the file names, which can make it impossible to distinguish between different files.
For instance, when compiling two files `a.swift` and `b.swift`, both containing the code `"private func foo() { }"`, with the command `"$ swiftc b.swift a.swift -dump-ast"`, we get this output:
```
(source_file
  (func_decl "foo()" interface type='() -> ()' access=private
    (parameter_list)
    (brace_stmt)))
(source_file
  (func_decl "foo()" interface type='() -> ()' access=private
    (parameter_list)
    (brace_stmt)))
```
It would be very useful if this information included the file name (or, even better, the file path). Something like this:
```
(source_file "/path/to/a.swift"
  (func_decl "foo()" interface type='() -> ()' access=private
    (parameter_list)
    (brace_stmt)))
(source_file "/path/to/b.swift"
  (func_decl "foo()" interface type='() -> ()' access=private
    (parameter_list)
    (brace_stmt)))
```

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-7904](https://bugs.swift.org/browse/SR-7904).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

fedetrim committed:
[SR-7904] Add source filenames in the output of `-dump-ast`

brentdax commented:
This looks like a great start! It looks like it doesn't quote the path, though; I think that would be better, since other places where we put an arbitrary string into the AST usually quote them.

fedetrim commented:
@brentdax you are right! Fixed that, and also removed my `FileColor` in favour of the already existing `LocationColor`. Let me know if that's OK or another color is needed/wanted here. Thanks!

jckarter commented:
@swift-ci Please smoke test

fedetrim commented:
@jckarter @brentdax hey guys! I see the smoke tests passed successfully, is there anything else I can do to improve this PR?

jckarter commented:
Looks good, thanks for reminding me to commit it!
